autoscheduler/frontend/src/components/SchedulingPage/Schedule/Schedule.tsx

```diff
@@ -399,6 +421,45 @@ const Schedule: React.FC = () => {
     </div>
   ));
 
+  // When term is chagned, fetch saved availabilities for the new term
+  React.useEffect(() => {
+    if (term) {
+      fetch(`sessions/get_saved_availabilities?term=${term}`).then((res) => (
+        res.json()
+      )).then((avails: Availability[]) => {
+        // We're done loading - hide the loading indicator and set the new availabilities
+        setIsLoadingAvailabilities(false);
+        dispatch(setAvailabilities(avails));
+      });
+    }
+
+    // on unmount, clear availabilities
+    return (): void => {
+      dispatch(clearAvailabilities());
+    };
```

Comment:
Is this `clearAvailabilties` call necessary? I think I did it cause Ryan did something similar with course cards, but I honestly couldn't remember why it would be useful. Maybe for changing terms with #349?

Reply:
In course cards it was so they didn't carry over when you went back to the main page, I don't see why not to do it here as well